src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/visits/PrisonVisitsService.kt

```diff
@@ -88,20 +98,25 @@ class PrisonVisitsService(
     val mappingDto =
       mappingService.getMappingGivenVsipId(visitCancelledEvent.visitId)
         ?: throw ValidationException("No mapping exists for VSIP id ${visitCancelledEvent.visitId}")
-          .also { telemetryClient.trackEvent("visit-cancelled-failed", telemetryProperties) }
+          .also { telemetryClient.trackEvent("visit-cancelled-mapping-failed", telemetryProperties) }
+
+    val mapWithNomisId = telemetryProperties.plus(Pair("nomisVisitId", mappingDto.nomisId))
 
-    nomisApiService.cancelVisit(
-      CancelVisitDto(
-        offenderNo = visitCancelledEvent.prisonerId,
-        nomisVisitId = mappingDto.nomisId,
-        outcome = "VISCANC" // TODO mapping
+    try {
+      nomisApiService.cancelVisit(
+        CancelVisitDto(
+          offenderNo = visitCancelledEvent.prisonerId,
+          nomisVisitId = mappingDto.nomisId,
+          outcome = "VISCANC" // TODO mapping
+        )
       )
-    )
+    } catch (e: Exception) {
+      telemetryClient.trackEvent("visit-cancelled-failed", mapWithNomisId)
```

Comment:
hmm, not sure this is required, won't it just retry? in which case a telemetry event isn't necessary perhaps? It will get logged anyway.

I wonder if telemetry events just reserved for permanent events we care about, so if there is a 503 and then it works 100ms later this would write a telemetry event that could perhaps be confusing

Reply:
good point, Ill remove it